ci/lava/tests/test-component-update.py

```diff
@@ -129,8 +138,35 @@ def test_get_pre_update_testinfo(
                         (image_data["image_name"], item["args"])
                     )
                 elif item["test_type"] == "app_bank_compare":
+                    # Work out the version of the currently installed
+                    # application. If there isn't an application installed
+                    # then nothing will be found and the expected_version
+                    # will be set to 0.
+                    #
+                    # We look in the output for a line with the format:
+                    #   "bundle": "/home/app/sample-app-5-good/1"
+                    # and extract the "1"
+
+                    app_info = get_app_info(
+                        item["args"]["app_name"],
+                        TestComponentUpdate.dut_address,
+                        execute_helper,
+                        app_output=True,
+                        raise_on_error=False,
+                    )
+                    expected_version = 0
+                    for line in app_info.splitlines():
+                        if '"bundle":' in line:
+                            bundle_values = re.compile('"[^"]*"').findall(line)
```

Comment:
Sorry, should've metioned this earlier but... I think it might be better to put this `appinfo` output parsing into a helper function which takes a single `appinfo` parameter and returns the `expected_version`, so the other temporary variables aren't polluting this execution context. Having a named function will also make it a bit easier to understand this code.

Reply:
I have created a new helper function.